File: sliver/11501/11501.js
// TimeComplexity: O(n)
// SpaceComplexity: O(1)
let input = require("fs")
  .readFileSync(__dirname + "/input.txt")
  .toString()
  .trim()
  .split("\n");

const testCase = Number(input[0]);
let lineIndex = 1;

for (let t = 0; t < testCase; t++) {
  const n = Number(input[lineIndex]);

  const prices = input[lineIndex + 1].split(" ").map(Number);

  let maxPrice = 0;
  let totalProfit = 0;

  for (let i = n - 1; i >= 0; i--) {
    const currentPrice = prices[i];

    if (currentPrice > maxPrice) {
      maxPrice = currentPrice;
    } else {
      totalProfit += maxPrice - currentPrice;
    }
  }

  console.log(totalProfit);
  lineIndex += 2;
}

File: sliver/11501/11501.py
import sys

input = sys.stdin.read().strip().split("\n")

input = [list(map(int, line.split())) for line in input]
test_case = input[0][0]
test_case_index = 1

for i in range(test_case):
    n = input[test_case_index]
    arr = input[test_case_index + 1]
    max_price = 0
    total_profit = 0
    for j in range(len(arr) - 1, -1, -1):
        current_price = arr[j]
        if current_price > max_price:
            max_price = current_price
        else:
            total_profit += max_price - current_price

    print(total_profit)

    test_case_index += 2
